Automattic/pocket-casts-android
# Bookmarks -  Replace bookmark creation sound file + add platform check

## Description

Part of https://github.com/Automattic/pocket-casts-android/issues/1307

This replaces bookmark creation sound file taken from the [iOS](https://github.com/Automattic/pocket-casts-ios/pull/1076/files#diff-95592ff28f73019c1e0ed591d742daa94d658fdcc34d9ea0b8859f5c8cf83664 ) app.

This also adds a platform check preventing a foreground bookmark screen when adding a bookmark using headset controls. That means, if bookmarks are enabled for a platform other than phone (like `Android Auto`), they will be added with the default "Bookmark" title in the background without showing the "Add bookmark" foreground screen.

## Testing Instructions

#### Phone 

1. Install debug build
2. Launch the app 
3. Login to a Plus account
4. Go `Profile` tab -> `Settings` -> `Headphone controls`
5. Tap on "Previous action" 
6. Select "Add bookmark" option
7. Tap on "Next action" 
6. Select "Add bookmark" option
8. Go to "Discover" tab and play any episode
9. Connect headphone to the device 
10. With the app backgrounded, perform skip previous action through headphone (double tap if Apple pods or Pixel pro buds)
    [Or you can run command `adb shell input keyevent 88` to emulate the action]
14. ✅ Notice that a notification is shown that the bookmark is added and sound is played
15. Perform skip next action through headphone (triple tap if Apple pods or Pixel pro buds)
    [Or you can run command `adb shell input keyevent 87` to emulate the action]
16. ✅ Notice that a notification is shown that the bookmark is added and sound is played

#### Android Automotive/ Wear OS

- Play an episode on Android Automotive and Wear OS platforms
- Connect headphone to the device  and perform skip previous
   [Or you can run command `adb shell input keyevent 88` to emulate the action]
- ✅ Notice that bookmark is not added through headset controls on these platforms

#### Android Auto

> **Warning**
> Test with caution

This takes bookmark-enabled settings from the phone app. So that a bookmark is added without showing the "Add bookmark" foreground screen requiring user interaction on Android Auto, I added a `Util.isCarUiMode(context)` check in dabdad15fd185c11a32a31d2aa42002858779a76

- Makes changes in the code to return true from `Util.isCarUiMode(context)`
- Add a log [before line number 40](https://github.com/Automattic/pocket-casts-android/commit/dabdad15fd185c11a32a31d2aa42002858779a76#diff-dae5888925540a8e4cfaab3c3752b465b566f4495801f732a7871bed2d41b858R40) to check value for `context.isAppForeground()`
- Enable bookmarks for phone app and enable "Add bookmark" for headphone controls "Skip previous" action 
- Play an episode and background the app on phone
- Connect and run the [DHU](https://developer.android.com/training/cars/testing/dhu)
- Run command `adb shell input keyevent 88` on terminal to emulate the "Skip previous" action 
- Notice in logs that the app is in the foreground
- ✅ Notice that the bookmark is added with the default "Bookmark" title 

## Checklist
 
#### I have tested any UI changes...
